cli/src/config.rs

```diff
@@ -17,33 +18,79 @@ To learn more, checkout https://apollo.dev/cli/telemetry\n";
 #[derive(Debug, Serialize, Deserialize)]
 pub struct CliConfig {
     pub machine_id: String,
+    // This can be used for a service api key, or a personal api key.
+    // The CLI doesn't differentiate at the moment.
+    pub api_key: Option<String>,
+}
+
+fn save(path: &PathBuf, cli_config: &CliConfig) -> Result<(), Box<dyn Error + 'static>> {
+    let toml = toml::to_string(cli_config).unwrap();
+
+    fs::create_dir_all(&path.parent().unwrap())?;
+    debug!("Wrote cli config to path {}", path.to_str().unwrap());
```

Comment:
this line is before we actually write, which will be misleading if we fail writing. Maybe phrase as "Writing CLI config ..." ?